packages/blockchain-extension/extension/commands/addEnvironmentCommand.ts

```diff
@@ -286,4 +241,128 @@ export async function addEnvironment(): Promise<void> {
+                            return;
+                        }
+                    } else {
+                        // There is a connection problem other than certificate related. Throw error.
+                        throw new Error(`Unable to connect to the IBM Blockchain Platform network: ${errorWithoutAuth.message}`);
+                    }
+                }
+            }
+        }
+        // We will now atempt to connect using key and secret. If it fails, there is a problem with the key and secret provided.
+        try {
+            requestOptions.auth = { username: userAuth1, password: userAuth2 };
+            await Axios.get(api, requestOptions);
+        } catch (errorConnecting) {
+            throw new Error(`Problem detected with the authentication information provided: ${errorConnecting.message}`);
+        }
+        // Securely store API key and secret
+        try {
+            await keytar.setPassword('blockchain-vscode-ext', url, `${userAuth1}:${userAuth2}:${requestOptions.httpsAgent.options.rejectUnauthorized}`);
+        } catch (errorStorePass) {
+            throw new Error(`Unable to store the required credentials: ${errorStorePass.message}`);
+        }
+
+        return url;
+    }
+
+    async function getOpsToolsAccessInfoSaaS(): Promise<string[]> {
+        const ibpResources: IBlockchainQuickPickItem<string>[] = [];
+        const accessToken: string = await ExtensionsInteractionUtil.cloudAccountGetAccessToken(true);
+        if (!accessToken) {
+            return;
+        }
+        const requestOptions: any = { headers: { Authorization: `Bearer ${accessToken}` } };
+        const baseUrl: string = 'https://resource-controller.cloud.ibm.com';
+        let resourcesUrl: string = `${baseUrl}/v2/resource_instances`;
+        while (resourcesUrl) {
+            const response: any = await Axios.get(resourcesUrl.toString(), requestOptions);
+
+            for (const resource of response.data.resources) {
+                if ( resource.resource_plan_id === 'blockchain-standard' ) {
+                    ibpResources.push({
+                        label: resource.name,
+                        description: resource.guid,
+                        data: resource
+                    });
+                }
+            }
+            if (response.data.next_url) {
+                resourcesUrl = `${baseUrl}${response.data.next_url.toString()}`;
+            } else {
+                break;
+            }
+        }
+
+        let chosenIbp: IBlockchainQuickPickItem<any>;
+        if (ibpResources.length === 0) {
+            throw new Error('There are no IBM Blockchain Platform resources associated with the chosen account');
+        } else if (ibpResources.length === 1) {
+            chosenIbp = ibpResources[0];
+        } else {
+            chosenIbp = await UserInputUtil.showQuickPickItem('Please select the IBM Blockchain Platform to connect to', ibpResources) as IBlockchainQuickPickItem<any>;
```

Comment:
Is this message clear enough for a user - would `Please select the IBM Blockchain Platform resource to connect to` be clearer?

Reply:
Just checked the design and actually the message should be 'Select an IBM Blockchain Platform service instance', so I'll change it to that.

Reply:
Also changed the wording of the previous error to say 'instances' instead of 'resources'